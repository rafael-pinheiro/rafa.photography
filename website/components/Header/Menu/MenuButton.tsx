import styles from './MenuButton.module.scss';

export default function MenuButton({ onClick } : { onClick: () => void}) {
  return (
    <button
      className = {styles.button}
      onClick = {onClick}
    >
      <span className={styles.label}>menu</span>
      <span className={styles.hamburguer}>
        <span className={styles.line} />
        <span className={styles.line} />
        <span className={styles.line} />
      </span>
    </button>
  )
}
