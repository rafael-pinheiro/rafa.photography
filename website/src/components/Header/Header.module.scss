.header {
  @apply flex items-center justify-between p-12;
}
