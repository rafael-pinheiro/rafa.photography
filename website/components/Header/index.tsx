import { useState } from 'react';
import styles from './Header.module.scss';
import MenuButton from './Menu/MenuButton';
import Menu from './Menu/Menu';

export default function Header() {
  const [ open , setOpen ] = useState(false);

  return (
    <div className={`${styles.header} ${open && 'open'}`}>
      <h1>Rafa</h1>
      <Menu />
      <MenuButton onClick={() => setOpen(!open)} />
    </div>
  );
}
