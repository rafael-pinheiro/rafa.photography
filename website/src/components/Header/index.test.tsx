import renderer from 'react-test-renderer';
import Header from './index';
import MenuButton from './Menu/MenuButton';


it('renders correctly', () => {
  const component = renderer.create(<Header />);

  expect(component.toJSON()).toMatchSnapshot();

  const button = component.root.findByType(MenuButton);

  renderer.act(() => {
    button.props.onClick();
  });

  expect(component.toJSON()).toMatchSnapshot();
});
