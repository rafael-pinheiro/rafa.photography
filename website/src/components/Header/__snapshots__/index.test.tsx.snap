// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="undefined "
>
  <h1>
    Rafa
  </h1>
  <nav>
    <a
      href="/"
    >
      <span>
        Home
      </span>
      <small>
        Back to homepage
      </small>
    </a>
    <a
      href="/"
    >
      <span>
        About
      </span>
      <small>
        Learn more about the author
      </small>
    </a>
    <a
      href="/"
    >
      <span>
        Contact
      </span>
      <small>
        Get in touch
      </small>
    </a>
  </nav>
  <button
    onClick={[Function]}
  >
    <span>
      menu
    </span>
    <span>
      <span />
      <span />
      <span />
    </span>
  </button>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="undefined open"
>
  <h1>
    Rafa
  </h1>
  <nav>
    <a
      href="/"
    >
      <span>
        Home
      </span>
      <small>
        Back to homepage
      </small>
    </a>
    <a
      href="/"
    >
      <span>
        About
      </span>
      <small>
        Learn more about the author
      </small>
    </a>
    <a
      href="/"
    >
      <span>
        Contact
      </span>
      <small>
        Get in touch
      </small>
    </a>
  </nav>
  <button
    onClick={[Function]}
  >
    <span>
      menu
    </span>
    <span>
      <span />
      <span />
      <span />
    </span>
  </button>
</div>
`;
